fusestore/store/src/executor/action_handler_test.rs

```diff
@@ -65,6 +77,64 @@ async fn test_action_handler_do_pull_file() -> anyhow::Result<()> {
     Ok(())
 }
 
+struct D {
+    plan: CreateDatabasePlan,
+    want: common_exception::Result<CreateDatabaseActionResult>,
+}
+
+/// helper to build a D
+fn case_db(db_name: &str, if_not_exists: bool, want: common_exception::Result<u64>) -> D {
+    let plan = CreateDatabasePlan {
+        db: db_name.to_string(),
+        if_not_exists,
+        engine: DatabaseEngineType::Local,
+        options: Default::default(),
+    };
+    let want = match want {
+        Ok(want_db_id) => Ok(CreateDatabaseActionResult {
+            database_id: want_db_id,
+        }),
+        Err(err) => Err(err), // Result<i64,_> to Result<StoreDoActionResult, _>
+    };
+
+    D { plan, want }
+}
+
+struct T {
+    plan: CreateTablePlan,
+    want: common_exception::Result<CreateTableActionResult>,
+}
```

Comment:
Also, refactoring code within a `new-feature` PR may not be a good practice.

Another single PR that "Just move code, nothing really changed" is much easier to review than a PR with old codes mixed within new codes.

In this case, as you need to reuse some of the old codes, just copy-paste them to a new standalone mod level function and leaving a TODO for refactoring the duplicated stuff would be comfortable for readers.

Reply:
Got it